pyprof/nvtx/nvmarker.py

```diff
@@ -312,15 +395,29 @@ def wrapper_func(*args, **kwargs):
             # Pop trace marker
             nvtx.range_pop()
 
+        if config.capture_input_ops:
+            capture_outputs(call_id, result)
+            # Store the callid -> op_name mapping
+            if config.func_stack_enabled:
+                traceMarker_str = traceMarker_str.replace("\'", "\"")
+                traceMarker_dict = json.loads(traceMarker_str)
+                call_id_to_op_map[call_id] = traceMarker_dict['funcStack']
+                call_id = call_id + 1
```

Comment:
Not sure if we have a defined behavior if capture_input_ops is true + func_stack_enabled is false, but shouldn't call_id be incremented regardless of func_stack_enabled?

Reply:
You are correct, it is not really defined. The capture_input_ops flag makes the assumption that the function stack tracing code is enabled. I was planning to codify that once we do the refactor.

Reply:
Does it make that assumption? This very code is checking for func_stack_enabled. Wouldn't everything else work if you just unindented the call_id = call_id + 1?

Reply:
The call_id would still increment but none of the mapping for the input capture works without the function stack tracing to enable resolving the names.
What would you ultimately like the behavior to be here?

Reply:
Good point.  

One option is to go with the old code behavior, which I think was putting the source callID instead of the source name?

The easier option would be to require that func_stack_enabled be true if capture_input_ops is true. Add an assert in the config class. Then you don't have to worry about these cases.

Reply:
I set the enable_function_stack flag if either option is set right now. That makes the most sense right now.